### ISSUE TITLE
Feature request   FileCreateDate

### ISSUE THREAD

**Issue by Philshappy** [FrankBijnen/ExifToolGui#545]
Would it be possible to modify the FileCreateDate just like the FileModifyDate and add the feature for both to be either Exif:DateTimeOriginal or an option to enter in a custom date.

**Comment by FrankBijnen** [comment]
@Philshappy 

Hi Phil,

Will look into it. Give me some time

Frank

**Comment by FrankBijnen** [comment]
Hi Phil,

To make sure we're talking about the same.

- You want to change FileCreateDate  that shows in Windows Explorer? (The dates in the Exif metadata are OK)

- I know 2 functions that will change the FileModifyDate based on the metadata. Do you want me to change those and add FileCreateDate?


1) Modify/Exif date time shift
![mod1](https://github.com/user-attachments/assets/79cf49fe-2cce-401f-bc28-afe35ba78c8d)

![mod2](https://github.com/user-attachments/assets/ac5b45e3-7925-4a38-b507-738777a78b32)

2) Various/File date modified as in Exif
![var1](https://github.com/user-attachments/assets/9fb73867-138f-4fbb-aa2c-23c013f57d1f)

![var2](https://github.com/user-attachments/assets/f4f7ee76-9220-430d-bffb-a4d91280f012)

Meanwhile you could try these ExifTool direct commands. (Save them as predefined!)

`-FileCreateDate<Exif:DateTimeOriginal -FileModifyDate<Exif:DateTimeOriginal`

`-FileCreateDate="2022:09:20 10:13:26" -FileModifyDate="2022:09:20 10:13:26"`

![etdirect](https://github.com/user-attachments/assets/6853b733-ce18-4963-990a-ba378a0c07fd)

Frank

**Comment by Philshappy** [comment]
I was referring to the function 'Various.File:Date modified as in Exif'

This function modifies the windows explorer file modified date. Would it be possible to add the windows explorer file create date.  Could the new dates come from either the exif dates as you already have or an input box for a custom date that can be entered by the user,

Also, can the 'Modify.ExifDateTime equalize' function modified to add an option for a custom input date and time that can be entered by the user instead of only exif dates.

Phil

________________________________

**Comment by FrankBijnen** [comment]
All clear now Phil.

Will let you know when I have something ready.

Frank

**Comment by FrankBijnen** [comment]
HI Phil,

Updates available in the pre-rel: https://1drv.ms/f/s!AhVJC-QzM33pknOS7KwBi09lgNql?e=n7ztFX

- Various/File data created and modified ....
I only added the Date created.  I'm thinking of removing this function, because you can do all with Modify'Exif: Date time shift
![Various_file](https://github.com/user-attachments/assets/4944a51d-2bec-4de7-aaa1-592b16262e2b)

- Modify/Exif: Date time shift
I have added the Checkbox **Change file: Date created**
**Date time result** Has been split in to 2 parts, the Date and the Time,  and has become editable. Tthe Date part has a calender,.
When you change the field, it calculates the **Shift Amount.** Effectively you can specify what the Date time should be, or specify a Shift Amount.

![Modify_shift](https://github.com/user-attachments/assets/051c34aa-39b1-4082-a2ca-49107cbbbe44)

![Modify_shift2](https://github.com/user-attachments/assets/527f06d5-e9c3-4fc6-803e-72c8d37a93d0)

Notes:
- If you only want to modify the file dates, you can leave the **Shift Amount** to zeroes, and just check the Change file checkboxes. This will do the same as "Various/File date created and modified.."

- Multi-selecting: The shift amount is still applied to each selected file. If selected files have different dates/times before, then they will different after.

Frank

**Comment by Philshappy** [comment]
Frank,

The rename changes adds a lot of flexibility to changing the file names. Thanks.
      1) When I tried the Remove leading Date/Time it appears to remove whatever is leading so if I renamed the file to 'Car 20240101' instead of removing the date/time it removes the car.
      2) When file names result like 24-01-01-1 is it possible to put a space after the date/time so 24-01-01 -1 or some other delimiter to separate the name from the unique counter.

Comments on the Exif Time Shift
      1) Consider changing the name from Exif DateTime shift to something more descriptive of the full capability as you added file name changes. Maybe 'Date/Time Change'.
      2) The Shift amount and Date Time result to not always track. For example, I edit the date result directly in the text box with 2024:02:02 and the result display the previous date and the following 0078:06:22 00:00:00 shift amount. When I check the increment or decrement the result is a really wrong date.  Similar behavior occurs if I use the down arrow and scroll thru the calendar to select a new date.


Phil


________________________________

**Comment by FrankBijnen** [comment]
Hi Phil,

For the date time shifting I may have a solution for your request to only shift the Modifydate.

If you set the checkboxes like this (Date Time original checked) the shift amount and resulting date are calculated relative to the Date Time original
![datetimeshift1](https://github.com/user-attachments/assets/2a48c2e8-1c07-423a-ad41-51cb6474e0ae)

If you set them like this (Only ModifyDate checked) the shift amount and resulting date are based on the Modify date.
![datetimeshift2](https://github.com/user-attachments/assets/05d592a6-4ead-4a00-8192-312dd642a874)

I think that makes sense, and your problem is solved.

Furthermore I changed the updating of the filedates.
- Prior to my change both the Created date and Modified date were copied from the Date time original.
- Now I have changed it so that the Created date gets the date from exif:CreateDate and Modified date from exif:ModifyDate

I thought that was more logical.

Frank

P.S. 
All this is available in the pre-rel.
The next few days I have plans that dont include ExifToolGui, so I will not be very responsive.
